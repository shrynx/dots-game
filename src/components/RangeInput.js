// @flow
import styled from 'styled-components';
import { PURPLE, GREY } from '../styles/colors';

type Props = {
  value: number,
};

const RangeInput = styled.input`
  appearance: none;
  width: 85%;
  height: 30px;
  margin-bottom: 10px;
  &:focus {
    outline: none;
  }
  &::-webkit-slider-runnable-track {
    -webkit-appearance: none;
    background: ${({ value }: Props) =>
      `-webkit-linear-gradient(left,${PURPLE} ${(value - 0.5) * 10}%, ${GREY} 0%`});
    height: 5px;
    border-radius: 3px;
  }
  &::-webkit-slider-thumb {
    appearance: none;
    background-color: ${PURPLE};
    border-radius: 50%;
    height: 20px;
    width: 20px;
    max-width: 80px;
    position: relative;
    bottom: 8px;
    transition: border 1000ms ease;
    cursor: -webkit-grab;
  }
  &::-webkit-slider-thumb:active {
    height: 30px;
    width: 30px;
    bottom: 12px;
    cursor: -webkit-grabbing;
  }
`;

export default RangeInput;
