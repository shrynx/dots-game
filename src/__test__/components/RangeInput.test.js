import React from 'react';
import { render } from 'react-testing-library';
import RangeInput from '../../components/RangeInput';

it('RangeInput component should render correctly', () => {
  const { container } = render(<RangeInput value={5} onChange={() => {}} />);
  expect(container).toMatchSnapshot();
});
