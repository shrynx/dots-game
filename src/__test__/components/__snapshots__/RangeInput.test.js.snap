// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput component should render correctly 1`] = `
<div>
  <input
    class="sc-bdVaJa hBoKdj"
    value="5"
  />
</div>
`;
